addons/udes_sale_stock/models/stock_picking.py

```diff
@@ -0,0 +1,23 @@
+from odoo import models
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def warn_picking_any_previous_pickings_not_complete_sale(self):
+        """Return a warning string when not all pickings have progressed
+        to the same picking type or later.
+
+        Determine if this is the case by examining if all picking of the same picking type are
+        assigned, done or cancelled.
+        """
+        accepted_states = ["assigned", "done", "cancel"]
+        all_pickings = self.mapped("move_lines.sale_line_id.order_id.picking_ids")
+        pickings = all_pickings.filtered(
+            lambda x: x.picking_type_id in self.mapped("picking_type_id")
+        )
+        unaccepted_pickings = pickings.filtered(lambda x: x.state not in accepted_states)
```

Comment:
nit: you could use any character for your lambda functions like 'p' for picking instead of 'x'

Reply:
As an ex physicist, x is my favourite thing to call variables. Changed to p anyway